src/delagent/ui/admin-upload-delete.php

```diff
@@ -114,27 +115,84 @@ function Delete($uploadpk, $Depends = NULL)
   } // Delete()
 
   /**
+   * @param $uploadpks
+   * @brief starts deletion and handles error messages
+   * @return string
+   */
+  function initDeletion($uploadpks)
+  {
+    $V = "";
+    $errorMessages = [];
+    $deleteResponse = NULL;
+    for($i=0; $i < sizeof($uploadpks); $i++)
+    {
+      $deleteResponse = $this->TryToDelete(intval($uploadpks[$i]));
+
+      if($deleteResponse->getDeleteMessageCode() != DeleteMessages::SUCCESS)
+      {
+        $errorMessages[] = $deleteResponse;
+      }
+    }
+
+    if(sizeof($uploadpks) <= 0)
```

Comment:
It would be cleaner to move that to the top of the function. That is where one expects it.